mesonbuild/interpreter.py

```diff
@@ -2017,35 +1968,32 @@ def _process_script_args(
 
     @FeatureNewKwargs('add_install_script', '0.57.0', ['skip_if_destdir'])
     @permittedKwargs({'skip_if_destdir'})
-    def add_install_script_method(self, args: 'T.Tuple[T.Union[str, mesonlib.File, ExecutableHolder], T.Union[str, mesonlib.File, CustomTargetHolder, CustomTargetIndexHolder, ConfigureFileHolder], ...]', kwargs):
-        if len(args) < 1:
-            raise InterpreterException('add_install_script takes one or more arguments')
+    @typed_pos_args('meson.install_script', (str, mesonlib.File, ExecutableHolder, ExternalProgramHolder), varargs=(str, mesonlib.File, CustomTargetHolder, CustomTargetIndexHolder, ConfigureFileHolder, ExecutableHolder, ExternalProgramHolder))
+    def add_install_script_method(self, args: T.Tuple[T.Union[str,mesonlib.File, ExternalProgramHolder, ExecutableHolder], T.List[T.Union[str, mesonlib.File, CustomTargetHolder, CustomTargetIndexHolder, ConfigureFileHolder, ExecutableHolder, ExternalProgramHolder]]], kwargs: T.Dict[str, T.Any]) -> None:
         if isinstance(args[0], mesonlib.File):
             FeatureNew.single_use('Passing file object to script parameter of add_install_script',
                                   '0.57.0', self.interpreter.subproject)
         skip_if_destdir = kwargs.get('skip_if_destdir', False)
         if not isinstance(skip_if_destdir, bool):
             raise InterpreterException('skip_if_destdir keyword argument must be boolean')
-        script_args = self._process_script_args('add_install_script', args[1:], allow_built=True)
+        script_args = self._process_script_args('add_install_script', args[1], allow_built=True)
```

Comment:
Just reading randomly the diff to see how it works. I don't think that change is correct, it process only the first arg instead of all of them, no?

Reply:
Forget that, I understood now, you change args to be a tuple.